### ISSUE TITLE
problem building docs with ford 5.0.4

### ISSUE THREAD

**Issue by ilovezfs** [jacobwilliams/json-fortran#225]
I tried upgrading the Homebrew formula to 5.1.0, but
```
-- Running FORD to dynamically compute documentation outputs, this could take a while...
Traceback (most recent call last):
  File "/usr/local/Cellar/ford/5.0.4/libexec/lib/python2.7/site-packages/ford/output.py", line 98, in __init__
    self.docs.append(FilePage(data,project,item))
  File "/usr/local/Cellar/ford/5.0.4/libexec/lib/python2.7/site-packages/ford/output.py", line 201, in __init__
    self.html = self.render(data,proj,obj)
  File "/usr/local/Cellar/ford/5.0.4/libexec/lib/python2.7/site-packages/ford/output.py", line 326, in render
    return template.render(data,src=obj,project=proj)
  File "/usr/local/Cellar/ford/5.0.4/libexec/vendor/lib/python2.7/site-packages/jinja2/environment.py", line 989, in render
    return self.environment.handle_exception(exc_info, True)
  File "/usr/local/Cellar/ford/5.0.4/libexec/vendor/lib/python2.7/site-packages/jinja2/environment.py", line 754, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/Cellar/ford/5.0.4/libexec/lib/python2.7/site-packages/ford/templates/file_page.html", line 1, in top-level template code
    {% extends "base.html" %}s
  File "/usr/local/Cellar/ford/5.0.4/libexec/lib/python2.7/site-packages/ford/templates/base.html", line 138, in top-level template code
    {% block body %}
  File "/usr/local/Cellar/ford/5.0.4/libexec/lib/python2.7/site-packages/ford/templates/file_page.html", line 12, in block "body"
    {{ macros.info_bar(src.name,src.meta,project_url,src.hierarchy,src.lines_description(project.file_lines)) }}
  File "/usr/local/Cellar/ford/5.0.4/libexec/lib/python2.7/site-packages/ford/sourceform.py", line 231, in lines_description
    description = "{:4.1f}% of total for {}.".format(float(self.num_lines)/total*100,pretty_obj[obj])
ZeroDivisionError: float division by zero
Error encountered.
-- Done dynamically computing FORD outputs.
CMake Error at CMakeLists.txt:199 (add_custom_command):
  add_custom_command Wrong syntax.  A TARGET or OUTPUT must be specified.
```

If I use ford 4.6.2 instead, this doesn't occur.

**Comment by ilovezfs** [comment]
full jenkins log https://gist.github.com/ilovezfs/87ab46ac6b6d2a9edf7f119dcac1e1ff

**Comment by jacobwilliams** [comment]
Hmmm... It seemed to work with the latest ford on [this build](https://travis-ci.org/jacobwilliams/json-fortran/jobs/152219979), but that was using the `build.sh` script. Could it be a CMake issue of some kind? @zbeekman 

Also note: [this ford bug report](https://github.com/cmacmackin/ford/issues/160) does mention a division by zero error.

**Comment by zbeekman** [comment]
I'll try to take a look. Just back from vacation.

**Comment by ilovezfs** [comment]
@jacobwilliams @zbeekman any progress on fixing this with cmake?

**Comment by zbeekman** [comment]
The difference may be due to some complicated logic to extract file dependencies at CMake/configure time or due to passing different pre-processor defines.

**Comment by ilovezfs** [comment]
@zbeekman yeah. It's actually possibly worse than it looks since the proximate cause of the division by zero thing is only subsequent to a reraise by jinja2, so it's probably a bit of a red herring.

**Comment by ilovezfs** [comment]
@jacobwilliams @zbeekman ford 5.0.5 has fixed the problem https://github.com/cmacmackin/ford/issues/160#issuecomment-247359591 and I just shipped json-fortran 5.1.0 (https://github.com/Homebrew/homebrew-core/pull/3897) in https://github.com/Homebrew/homebrew-core/commit/98655e85d5f9d67100a720911857eb329292801c.
